lib/load-adapter-from-app-dependencies.js

```diff
@@ -39,18 +39,12 @@ module.exports = function loadAdapterFromAppDependencies(adapterPackageName, dat
   //  └─  └  ┴└─└─┘┴ ┴  ┴ ┴┴  ┴    ┘└┘└─┘─┴┘└─┘────┴ ┴└─┘─┴┘└─┘┴─┘└─┘└─┘  └  └─┘┴─┘─┴┘└─┘┴└─  ─┘
 
   // Since it is unknown so far, try and load the adapter from `node_modules`
-  sails.log.verbose('Loading adapter (`%s`) from this app\'s `node_modules/` directory...', adapterPackageName);
-
-  // Before trying to actually require the adapter, determine the path to the module
-  // relative to the app we're loading:
-  var userlandDependenciesPath = Path.resolve(sails.config.appPath, 'node_modules');
-  var adapterPackagePath = Path.join(userlandDependenciesPath, adapterPackageName);
 
 
   // Now try to require the adapter from userland dependencies (node_modules of the sails app).
   var adapter;
   try {
-    adapter = require(adapterPackagePath);
+    adapter = require(adapterPackageName);
```

Comment:
@stanislavromanov If I'm understanding you correctly, there's still a problem (at least with the versions of Node and NPM I've used in the past). If code in `sails-hook-orm` (a dependency of `sails`, which is a dependency of your project) attempts to `require('sails-mysql')`, then Node will look at the dependencies of `sails-hook-orm` -- in its node_modules folder.  And like you're saying, it will continue to look at higher and higher level directories that have a package.json, attempting to find the dependency.  But for some reason or another, this just doesn't work, at least not on all versions of Node/NPM that we support (I started off trying to do it that way, and only changed it after it didn't work for some developers.  I had to do the same thing for generators, view engines, hooks, socket.io adapters, and connect session adapters.)

All that said, it may be that this is fixed now on all versions of NPM/Node in common use today (i.e. Node >=0.10 and NPM >= 2.0).  If that's the case, then we just need some more test cases that verify that this works (Travis can take care of testing on multiple versions of Node-- I'll make sure I push up a travis.yml file for this repo).  I'd love to be able to simplify this code throughout the framework!